### PR TITLE
Make Component.Group expand only if necessary

### LINKED ISSUE #1791: Grouping fixed-size Components produces unexpected behavior
Chart with an `Axis` at the bottom: http://jsfiddle.net/yzfauLe7/
screen shot 2015-04-03 at 1 29 49 pm

Chart with a `Label` at the bottom: http://jsfiddle.net/yzfauLe7/1/
screen shot 2015-04-03 at 1 30 25 pm

Chart with `Group` containing an `Axis` and a `Label`: http://jsfiddle.net/yzfauLe7/2/
screen shot 2015-04-03 at 1 31 14 pm

????

### PR DESCRIPTION
**Pull request by jtlan**
Now, if a Group contains only fixed-width Components, its width will
expand to fill the available space only if those Components do not
share the same x alignment. Same for height, based on y alignment.

Added "Mocks" module which contains classes for use in testing.

Wrote better tests for checking Group behavior based on contents.
Moved a misplaced test.

Close #1791.

### CODE REVIEW

**Comment by jtlan** [comment]
Marking this as [BLOCKER] because it interferes with desired behavior when merging `SelectionBoxLayer` with fixed-width and fixed-height `Component`s (like an `Axis` of any type).

**Comment by jtlan** [comment]
The intended behavior should produce sensible results in general, and also not break #1255...

File: src/components/label.ts [hunk 1]
```diff
@@ -34,34 +32,6 @@ export module Component {
...
-     * @param {string} alignment The new setting, one of `["top", "center",
-     * "bottom"]`. Defaults to `"center"`.
-     * @returns {Label} The calling Label.
-     */
-    public yAlign(alignment: string): Label {
-      var alignmentLC = alignment.toLowerCase();
-      super.yAlign(alignmentLC);
-      this._yAlignment = alignmentLC;
-      return this;
-    }
```
Comment (bluong) [inline src/components/label.ts]: Yay for code removal

File: test/core/componentGroupTests.ts [hunk 2]
```diff
@@ -164,35 +126,80 @@ describe("ComponentGroups", () => {
...
+
+    it("fixed-size Components, same x and same y alignments", () => {
+      var svg = generateSVG(SVG_WIDTH, SVG_HEIGHT);
+      var tall = new Mocks.FixedSizeComponent(SVG_WIDTH / 4, SVG_HEIGHT / 2);
+      var wide = new Mocks.FixedSizeComponent(SVG_WIDTH / 2, SVG_HEIGHT / 4);
+      var cg = new Plottable.Component.Group([ tall, wide ]);
+      cg.renderTo(svg);
+      var largestWidth = cg.components().map((c) => c.width()).sort((a, b) => b - a)[0];
+      assert.strictEqual(cg.width(), largestWidth, "takes on the width of widest fixed-size component");
+      var largestHeight = cg.components().map((c) => c.height()).sort((a, b) => b - a)[0];
```
Comment (bluong) [inline test/core/componentGroupTests.ts]: Instead use the _Util.max

File: test/mocks.ts [hunk 3]
```diff
@@ -0,0 +1,25 @@
+///<reference path="testReference.ts" />
+
+module Mocks {
```
Comment (bluong) [inline test/mocks.ts]: Consider maybe incorporating it in the Plottable module?

File: test/mocks.ts [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+  export class FixedSizeComponent extends Plottable.Component.AbstractComponent {
```
Comment (bluong) [inline test/mocks.ts]: Is there not some Javscript mocking infrastructure in place? sinon / etc.?

**Comment by bluong** [comment]
We should see if we can improve on this testing infrastructure.  There is the SinonJS library that I believe is used sparingly in our codebase that we can possibly use more to our advantage.  Would using that be a good idea?

**Comment by jtlan** [comment]
Sinon only creates fake functions, not objects. However, I think it might be better if we change the module name to "`Stubs`".

File: test/mocks.ts (hunk 3 above)
Comment (jtlan) [inline test/mocks.ts]: It doesn't belong there since it's test code, rather than part of the main library.

**Comment by jtlan** [comment]
I think I came up with a better way to do this.

**Pull request closed without merging**